src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -154,10 +154,16 @@ private static bool TryGetByteArrayFromMemoryStream(MemoryStream data, out byte[
 
             try
             {
-                buffer = data.GetBuffer();
-                return true;
+                if (PortableShim.MemoryStream.GetBuffer != null)
+                {
+                    buffer = (byte[])PortableShim.MemoryStream.GetBuffer.Invoke(data, null);
```

Comment:
@davkean yep.  Already done in the core-clr branch ;) 

https://github.com/dotnet/roslyn/blob/core-clr/src/Compilers/Core/Desktop/EncodedStringText.cs#L139